velox/exec/SetAccumulator.h

```diff
@@ -120,6 +207,19 @@ struct StringViewSetAccumulator {
   /// Stores unique non-null non-inline strings.
   Strings strings;
 
+  /// Size (in bytes) of the serialized string values (this includes inline and
+  /// non-inline) strings. This value also includes the bytes for serializing
+  /// the length value (base.kVectorSizeT) of the strings.
+  /// Used for computing serialized buffer size for spilling.
+  size_t stringSetBytes = base.kVectorSizeT;
+
+  /// When serializing the strings for spilling, they are written in order of
+  /// their indexes. 'offsets' represents the offset of the unique value at that
+  /// index from the beginning of the serialization buffer. These offsets are
+  /// maintained to easily copy the unique value at that position in the
+  /// serialization buffer.
+  std::vector<size_t> offsets;
```

Comment:
This vector can be very large and its memory usage is not accounted for (it is using malloc directly instead of going via MemoryPool). Most queries do not trigger spilling, but will have to incur the costs of maintaining this vector. Would it be possible to remove it?

Reply:
It struck me that if the index of each position is serialized along with its length and value, then its not required that we deserialize the accumulator contents in order of their indices. So then a maintenance of a offsets vector or sorting when serializing is not required. I think this is a reasonable trade-off. wdyt ?